cargo-shuttle/src/main.rs

```diff
@@ -179,4 +182,28 @@ impl Shuttle {
         let owned = locks.get(0).unwrap().file().try_clone()?;
         Ok(owned)
     }
+
+    fn run_tests(&self, no_test: bool) -> Result<()> {
+        if no_test {
+            return Ok(());
+        }
+
+        let config = cargo::util::config::Config::default()?;
+        let working_directory = self.ctx.working_directory();
+        let path = working_directory.join("Cargo.toml");
+
+        let compile_options = CompileOptions::new(&config, CompileMode::Test).unwrap();
+        let ws = Workspace::new(&path, &config)?;
+        let opts = TestOptions {
+            compile_opts: compile_options,
+            no_run: false,
+            no_fail_fast: false,
+        };
+
+        let err = cargo::ops::run_tests(&ws, &opts, &[])?;
+        match err {
+            None => Ok(()),
+            Some(_) => Err(anyhow!("To ignore all tests, pass the `--no-test` flag")),
+        }
```

Comment:
```suggestion
        cargo::ops::run_tests(&ws, &opts, &[])?.expect("Some tests failed. To ignore all tests, pass the `--no-test` flag");
        Ok(())
```
I'm not sure if the original code causes the failure message(s) to be lost.

Feel free to add a test to any of the examples if you want to test the output. It seems the `from_authorization()` and `to_token()` in the `rocket/authentication` might be your best options for creating a unit test.

You can also create an integration test for this function (or the run function) if you want

Reply:
The suggested code above would be the opposite to what's required from what I can see. The `run_tests()` returns `Some` if there are errors, `None` if not. There may be a more idiomatic way to resolve this though still?

Reply:
Of course. I wasn't paying proper attention :man_facepalming: